src/main/java/com/company/array/RepeatedDNASequences.java

```diff
@@ -0,0 +1,55 @@
+package com.company.array;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RepeatedDNASequences {
+    public List<String> findRepeatedDnaSequences(String s) {
+        List<String> result = new ArrayList<>();
+        if (s.length() < 11) {
+            return result;
+        }
+        Map<String, Integer> map = new HashMap<>();
+        StringBuilder str = new StringBuilder(s.substring(0, 10));
+        map.put(str.toString(), 1);
+        for (int r = 10; r < s.length(); r++) {
+            str.deleteCharAt(0);
+            str.append(s.charAt(r));
+            map.put(str.toString(), map.getOrDefault(str.toString(), 0) + 1);
```

Comment:
Can we try out without stringbuilder,  because lot of strings will be created in heap memory?

Reply:
Assigned str.toString() to a variable and reusing it